src/EntryPoints/SpecialNetwork.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace MediaWiki\Extension\Network\EntryPoints;
+
+use HTMLForm;
+use IncludableSpecialPage;
+use MediaWiki\Extension\Network\NetworkFunction\NetworkConfig;
+use Title;
+
+/**
+ * @psalm-suppress PropertyNotSetInConstructor
+ */
+class SpecialNetwork extends IncludableSpecialPage {
+
+	public function __construct() {
+		parent::__construct( 'Network' );
+	}
+
+	/**
+	 * @param string|null $subPage
+	 */
+	public function execute( $subPage ) : void {
+		$request = $this->getRequest();
+		$output = $this->getOutput();
+		$this->setHeaders();
+
+		$config = new NetworkConfig();
+		$params = [];
+
+		$params[] =
+			'pages=' .
+			strtr( $request->getText( 'pages', Title::newMainPage()->getPrefixedText() ), "\n", '|' );
+
+		$exclude = $request->getText( 'exclude' );
+		if ( $exclude !== '' ) {
+			$params[] = 'exclude=' . strtr( $exclude, "\n", ';' );
+		}
+
+		$class = $request->getText( 'class' );
+		if ( $class !== '' ) {
+			$params[] = 'class=' . $class;
+		}
+
+		$options = $request->getText( 'options', json_encode( $config->getOptions(), JSON_PRETTY_PRINT ) );
+		$params[] = 'options=' . $options;
+
+		$enableDisplayTitle = filter_var(
+			$request->getText( 'enableDisplayTitle', strval( $config->getDefaultEnableDisplayTitle() ) ),
+			FILTER_VALIDATE_BOOL,
+			FILTER_NULL_ON_FAILURE
+		);
+		$params[] = 'enableDisplayTitle=' . ( $enableDisplayTitle ? 'true' : 'false' );
+
+		$labelMaxLength = $request->getInt(
+			'labelMaxLength',
+			$config->getDefaultLabelMaxLength()
+		);
+		$params[] = 'labelMaxLength=' . strval( $labelMaxLength );
+
+		/**
+		 * @psalm-suppress PossiblyNullReference
+		 */
+		$html = ( new NetworkFunction( $config ) )->handleParserFunctionCall(
+			$output,
+			$output->getTitle()->getFullText(),
+			$params
+		);
```

Comment:
It is nice how you reused things here. The design is meant to do this on the UseCase level though. So both the parser function and the special page would create the UseCase and call it. The special page would not use the parser function. This means some of the wiring code will be duplicated, but as you noticed, not everything is the same. In cases where the wiring is more complex, this duplication could be reduced by extracting it into a factory. Probably not warranted here yet.

Reply:
Agreed. I did some refactoring to NetworkPresenter to facilitate reusing at the UseCase level. Let me know what you think. I am still not completely happy with the interaction with Extension.